### PR TITLE
CLOUDP-164969: Migrate Atlas CLI Organizations Client to v2 SDK Client

### PR DESCRIPTION
**Pull request by matt-condon**
Updating the migration of Atlas CLI Organizations commands to the auto-generated v2 atlas SDK

Added conditional template generation

Other options we considered were 
- Creating a new store for each v2 endpoint
- Creating a new implementation of a store interface for each v2 endpoint

We opted to go with the `interface` + conditional template generation as an approach for ease of integration.
We have many commands to migrate and test, and in many cases, very few changes will be required for this approach.

If/when we end up separating mongoCLI and atlasCLI at a later stage, the removal of the on-prem template and the `store isAtlas false` cases should be straightforward 

_Jira ticket:_ CLOUDP-164969

## Checklist

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added any necessary documentation in document requirements section listed in [CONTRIBUTING.md](https://github.com/mongodb/mongodb-atlas-cli/blob/master/CONTRIBUTING.md) (if appropriate)
- [ ] I have addressed the @mongodb/docs-cloud-team comments (if appropriate)
- [ ] I have updated [test/README.md](https://github.com/mongodb/mongodb-atlas-cli/blob/master/test/README.md) (if an e2e test has been added)
- [x] I have run `make fmt` and formatted my code

### CODE REVIEW

**Comment by matt-condon** [comment]
e2e tests passing:
https://spruce.mongodb.com/version/64182ec40305b97bd9d8d2b4/tasks?sorts=STATUS%3AASC%3BBASE_STATUS%3ADESC

File: internal/cli/iam/organizations/describe.go [hunk 1]
```diff
@@ -57,7 +63,7 @@ func (opts *DescribeOpts) Run() error {
 // mongocli iam organizations(s) describe <ID>.
 func DescribeBuilder() *cobra.Command {
 	opts := new(DescribeOpts)
-	opts.Template = describeTemplate
+	// opts.Template = describeTemplate
```
Comment (gssbzn) [inline internal/cli/iam/organizations/describe.go:66]:
```suggestion
```
let's clean up unused code

File: internal/cli/output_opts.go [hunk 2]
```diff
@@ -53,6 +53,22 @@ func (opts *OutputOpts) InitOutput(w io.Writer, t string) func() error {
...
+			opts.Template = cloudTemplate
+			opts.OutWriter = w
+			return nil
+		}
+	}
+	return func() error {
+		opts.Template = onPremTemplate
+		opts.OutWriter = w
+		return nil
+	}
```
Comment (gssbzn) [inline internal/cli/output_opts.go:69]:
```suggestion

	return func() error {
		if config.IsCloud() {
			opts.Template = cloudTemplate
		else {
			opts.Template = onPremTemplate
		}
		opts.OutWriter = w
		return nil
	}
```

File: internal/store/organizations.go [hunk 3]
```diff
@@ -44,6 +44,7 @@ type OrganizationDeleter interface {
 func (s *Store) Organizations(opts *atlas.OrganizationsListOptions) (*atlas.Organizations, error) {
 	switch s.service {
 	case config.CloudService, config.CloudGovService:
+		//TODO: Migrate once OrganizationsListOptions.IncludeDeletedOrgs property is generated CLOUDP-166105
```
Comment (gssbzn) [inline internal/store/organizations.go:47]: `IncludeDeletedOrgs` will never work for atlas

Comment (matt-condon) [reply internal/store/organizations.go:47]: Have the changes prepared but found another bug [CLOUDP-167160](https://jira.mongodb.org/browse/CLOUDP-167160)
Will raise in a follow up PR

File: internal/cli/iam/organizations/describe.go [hunk 4]
```diff
@@ -27,11 +27,17 @@ import (
...
 )
 
-const describeTemplate = `ID	NAME
-{{.ID}}	{{.Name}}
+const (
+	describeTemplateCloud = `ID	NAME
+{{.Id}}	{{.Name}}
 `
+	describeTemplateOnPrem = `ID	NAME
+{{.Id}}	{{.Name}}
```
[the remark this answers is not in the document]
Comment (matt-condon) [reply internal/cli/iam/organizations/describe.go]: Good spot, had uncommitted changes. Addressed 🙂

**Review by swattyT — APPROVED** [review]
LGTM!

File: internal/cli/iam/organizations/describe_test.go [hunk 5]
```diff
@@ -18,29 +18,48 @@
...
+		OutputOpts: cli.OutputOpts{
+			Template:  describeTemplateCloud,
+			OutWriter: buf,
+		},
 	}
+
 	if err := opts.Run(); err != nil {
 		t.Fatalf("Run() unexpected error: %v", err)
 	}
+	assert.Equal(t, `ID     NAME
```
Comment (wtrocki) [inline internal/cli/iam/organizations/describe_test.go:61]: IMHO this will be making tests to flaky. Would ignore assertions.  or use contains instead

**Comment by wtrocki** [comment]
Looks very good. from my side. Good to merge

**Review by gssbzn — APPROVED** [review]
LGTM

File: internal/cli/iam/organizations/describe_test.go (hunk 5 above)
Comment (matt-condon) [reply internal/cli/iam/organizations/describe_test.go:61]: Agree, this won't scale well as we migrate more stores and have to amend more templates

**Pull request merged**